blackboxopt/optimization_loops/sequential.py

```diff
@@ -94,16 +101,25 @@ def run_optimization_loop(
                 objectives=objectives,
                 catch_exceptions_from_evaluation_function=catch_exceptions_from_evaluation_function,
             )
+
             logger.info(
-                "Reporting the result from the evaluation function to the optimizer:\n"
-                + f"{json.dumps(evaluation.to_dict(), indent=2)}"
+                "Reporting the following evaluation result to the optimizer:\n%s",
+                json.dumps(
```

Comment:
As we here use it only for logging, might stringifying via [`pprint.pformat`](https://docs.python.org/3/library/pprint.html#pprint.pformat) be an option? Might hope would be, that it already handles most issues that can occur when serializing to json...

Reply:
With `str()` and `json.dumps` as currently implemented:
```
  "created_unixtime": 1686052139.35081,
  "context": null,
  "constraints": null,
  "finished_unixtime": 1686052139.351386,
  "stacktrace": null,
  "user_info": "{'muy': 'complex', 'lamda': <function evaluation_function.<locals>.<lambda> at 0x7f08e22c9790>, 'longlist': [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]}"
}
```

With `pprint.pformat(evaluation.to_dict())` instead of `json.dumps`
```
 'created_unixtime': 1686052409.133704,
  'finished_unixtime': 1686052409.134927,
  'objectives': {'loss': 1.0},
  'optimizer_info': {},
  'settings': {},
  'stacktrace': None,
  'user_info': { 'lamda': <function evaluation_function.<locals>.<lambda> at 0x7f0964e23b80>,
                 'longlist': [ 2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2,
                               2],
                 'muy': 'complex'}}
```

with `pprint.pformat()` just on the `user_info` and around `json.dumps`:
```
  "created_unixtime": 1686052342.399883,
  "context": null,
  "constraints": null,
  "finished_unixtime": 1686052342.400269,
  "stacktrace": null,
  "user_info": "{'lamda': <function evaluation_function.<locals>.<lambda> at 0x7f64ad4aa040>,\n 'longlist': [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],\n 'muy': 'complex'}"
}
```

To me the current implementation looks the cleanest, or am I missing specific pprint args?

Reply:
> or am I missing specific pprint args? 

well, `compact=True` should help, could you try that?

Probably it still wouldn't look exactly like the custom solution, on the other hand it also wraps lines after _n_ chars etc... But I see the simpler implementation as main benefit.

Reply:
With compact it does look a wee bit nicer
```
{'configuration': {'p': 0.40032391669228673},
 'constraints': None,
 'context': None,
 'created_unixtime': 1686055220.515068,
 'finished_unixtime': 1686055220.516517,
 'objectives': {'loss': 1.0},
 'optimizer_info': {},
 'settings': {},
 'stacktrace': None,
 'user_info': {'lamda': <function evaluation_function.<locals>.<lambda> at 0x7f9d4fda9820>,
               'longlist': [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
                            2, 2, 2],
               'muy': 'complex'}}
```

What do the others think, @aeivazi and @sfalkner? I'm open to changing to pprint, or has someone of you benefited from being able to copy valid JSON from the log output?

Reply:
> or has someone of you benefited from being able to copy valid JSON from the log output?

What difference do you see regarding that with `pformat()` vs. `json.dumps()`? By allowing non-serializable objects, both never will be valid JSONs...

Reply:
The difference is that the initially proposed `json.dumps` with `str(user_info)` results in valid JSON, it just changes the `user_info` value into a string, while pformat does not. But at least I haven't relied on that so far, just wanted to check because it's the only reason left for me why I'd consider json over pprint.

Reply:
Honestly, i have not yet used log output for anything else but additional information. I feel both proposal yields to the same result for me as a user. If `pformat` reduces number of code lines, i would vote for it.